src/main/webapp/app/detail-overview-list/exercise-detail.directive.ts

```diff
@@ -0,0 +1,50 @@
+import { ComponentRef, Directive, Input, OnDestroy, OnInit, ViewContainerRef } from '@angular/core';
+import { Detail, ShownDetail } from 'app/detail-overview-list/detail.model';
+import { DetailType } from 'app/detail-overview-list/detail-overview-list.component';
+import { TextDetailComponent } from 'app/detail-overview-list/components/text-detail.component';
+
+@Directive({
+    selector: '[jhiExerciseDetail]',
+    standalone: true,
+})
+export class ExerciseDetailDirective implements OnInit, OnDestroy {
+    @Input() detail: Detail;
+
+    private componentRef: ComponentRef<any>;
+
+    constructor(public viewContainerRef: ViewContainerRef) {}
+
+    ngOnInit() {
+        if (!this.isShownDetail()) {
+            return;
+        }
+        this.detail = this.detail as ShownDetail;
+
+        const detailTypeToComponent = {
+            [DetailType.Text]: TextDetailComponent,
+        };
+
+        const detailComponent = detailTypeToComponent[this.detail.type];
```

Comment:
I will add further components here in the future, this does not happen in this PR to keep the changeset manageable.